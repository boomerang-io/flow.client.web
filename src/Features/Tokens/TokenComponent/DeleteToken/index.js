import React from "react";
import { Button, ConfirmModal } from "@boomerang-io/carbon-addons-boomerang-react";
import { TrashCan16 } from "@carbon/icons-react";
import styles from "./deleteToken.module.scss";

function DeleteToken({ tokenItem, deleteToken }) {
  const message = tokenItem.toolTemplateName
    ? `If instances of ${tokenItem.toolTemplateName} are actively using this token, things will likely break.`
    : "If this token is actively being used, things will likely break.";
  return (
    <ConfirmModal
      modalTrigger={({ openModal }) => (
        <Button
          className={styles.button}
          iconDescription="deleteToken"
          kind="danger--ghost"
          onClick={openModal}
          renderIcon={TrashCan16}
          size="small"
          data-testid={`delete-token-button-${tokenItem.id}`}
        />
      )}
      affirmativeAction={() => deleteToken(tokenItem.id)}
      affirmativeButtonProps={{ kind: "danger" }}
      affirmativeText="Yes"
      negativeText="No"
      title={`Are you sure?`}
    >
      {message}
    </ConfirmModal>
  );
}

export default DeleteToken;
