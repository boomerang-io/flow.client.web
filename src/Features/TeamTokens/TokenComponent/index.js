import React, { useState } from "react";
import moment from "moment";
import cx from "classnames";
import { settings } from "carbon-components";
import {
  ButtonSkeleton,
  DataTable,
  DataTableSkeleton,
  Error404,
  ErrorMessage,
  Pagination,
} from "@boomerang-io/carbon-addons-boomerang-react";
import DeleteToken from "./DeleteToken";
import CreateToken from "./CreateToken";
import Header from "Components/Header";
import { TokensPermissions } from "Config/permissionsConfig";
import { useUserPermissions } from "Hooks";
import { arrayPagination, sortByProp } from "Utils/arrayHelpers";
import styles from "./tokensComponent.module.scss";

const { prefix } = settings;

const DEFAULT_PAGE_SIZE = 10;
const PAGE_SIZES = [DEFAULT_PAGE_SIZE, 20, 50, 100];

const headers = [
  {
    header: "Created By",
    key: "creatorName",
    sortable: true,
  },
  {
    header: "Description",
    key: "description",
    sortable: true,
  },
  {
    header: "Date Created",
    key: "creationDate",
    sortable: true,
  },
  {
    header: "Expires (UTC)",
    key: "expiryDate",
    sortable: true,
  },
  {
    header: "",
    key: "delete",
    sortable: false,
  },
];

const FeatureLayout = ({ children }) => {
  return (
    <div className={styles.container}>
      <Header title="Global Tokens" description="" />
      <div className={styles.content}>{children}</div>
    </div>
  );
};

function TokenComponent({ deleteToken, tokens, hasError, isLoading }) {
  const [page, setPage] = useState(1);
  const [pageSize, setPageSize] = useState(DEFAULT_PAGE_SIZE);
  const [sortKey, setSortKey] = useState("creationDate");
  const [sortDirection, setSortDirection] = useState("DESC");
  const deleteTokenAvailable = useUserPermissions()[TokensPermissions.deleteToken];

  const renderCell = (tokenItemId, cellIndex, value) => {
    const tokenDetails = tokens.find((token) => token.id === tokenItemId);
    const column = headers[cellIndex];
    switch (column.key) {
      case "creationDate":
      case "expiryDate":
        return (
          <p className={styles.tableTextarea}>
            {value ? moment(value).utc().startOf("day").format("MMMM DD, YYYY") : "---"}
          </p>
        );
      case "delete":
        return tokenDetails && tokenDetails.id && deleteTokenAvailable ? (
          <DeleteToken tokenItem={tokenDetails} deleteToken={deleteToken} />
        ) : (
          ""
        );
      default:
        return <p className={styles.tableTextarea}>{value || "---"}</p>;
    }
  };

  const handlePaginationChange = ({ page, pageSize }) => {
    setPage(page);
    setPageSize(pageSize);
  };

  function handleSort(e, { sortHeaderKey }) {
    const order = sortDirection === "ASC" ? "DESC" : "ASC";
    setSortKey(sortHeaderKey);
    setSortDirection(order);
  }

  const { TableContainer, Table, TableHead, TableRow, TableBody, TableCell, TableHeader } = DataTable;

  if (isLoading) {
    return (
      <FeatureLayout>
        <ButtonSkeleton className={styles.buttonSkeleton} small />
        <DataTableSkeleton
          data-testid="token-loading-skeleton"
          className={cx(`${prefix}--skeleton`, `${prefix}--data-table`, styles.tableSkeleton)}
          rowCount={DEFAULT_PAGE_SIZE}
          columnCount={headers.length}
          headers={headers.map((header) => header.header)}
        />
      </FeatureLayout>
    );
  }

  if (hasError) {
    return (
      <FeatureLayout>
        <ErrorMessage />
      </FeatureLayout>
    );
  }

  if (tokens) {
    return (
      <FeatureLayout>
        <div className={styles.tokenContainer}>
          <CreateToken />
        </div>
        {tokens.length > 0 ? (
          <>
            <DataTable
              rows={arrayPagination(tokens, page, pageSize, sortKey, sortDirection)}
              sortRow={(rows) => sortByProp(rows, sortKey, sortDirection.toLowerCase())}
              headers={headers}
              render={({ rows, headers, getHeaderProps }) => (
                <TableContainer>
                  <Table isSortable>
                    <TableHead>
                      <TableRow className={styles.tableHeadRow}>
                        {headers.map((header) => (
                          <TableHeader
                            id={header.key}
                            {...getHeaderProps({
                              header,
                              className: `${styles.tableHeadHeader} ${styles[header.key]}`,
                              isSortable: header.sortable,
                              onClick: handleSort,
                            })}
                            isSortHeader={sortKey === header.key}
                            sortDirection={sortDirection}
                          >
                            {header.header}
                          </TableHeader>
                        ))}
                      </TableRow>
                    </TableHead>
                    <TableBody className={styles.tableBody}>
                      {rows.map((row) => (
                        <TableRow key={row.id} >
                          {row.cells.map((cell, cellIndex) => (
                            <TableCell key={cell.id} style={{ padding: "0" }}>
                              <div className={styles.tableCell}>{renderCell(row.id, cellIndex, cell.value)}</div>
                            </TableCell>
                          ))}
                        </TableRow>
                      ))}
                    </TableBody>
                  </Table>
                </TableContainer>
              )}
            />
            <Pagination
              onChange={handlePaginationChange}
              page={page}
              pageSize={pageSize}
              pageSizes={PAGE_SIZES}
              totalItems={tokens.length}
            />
          </>
        ) : (
          <>
            <DataTable
              rows={tokens}
              headers={headers}
              render={({ headers }) => (
                <TableContainer>
                  <Table>
                    <TableHead>
                      <TableRow className={styles.tableHeadRow}>
                        {headers.map((header) => (
                          <TableHeader
                            key={header.key}
                            id={header.key}
                            className={`${styles.tableHeadHeader} ${styles[header.key]}`}
                          >
                            {header.header}
                          </TableHeader>
                        ))}
                      </TableRow>
                    </TableHead>
                  </Table>
                </TableContainer>
              )}
            />
            <Error404 title="No global tokens found" header={null} message={null} theme="boomerang" />
          </>
        )}
      </FeatureLayout>
    );
  }

  return null;
}

export default TokenComponent;
