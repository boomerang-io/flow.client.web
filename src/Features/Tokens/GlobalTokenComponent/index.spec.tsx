import React from "react";
import { fireEvent, screen } from "@testing-library/react";
import { tokens } from "ApiServer/fixtures";
import TokenComponent from "./index";

const props = {
  deleteToken: jest.fn(),
  tokens: tokens,
  hasError: false,
  isLoading: false,
};

describe("TokenComponent --- Snapshot", () => {
  it("Capturing Snapshot of TokenComponent", () => {
    const { baseElement } = global.rtlContextRouterRender(<TokenComponent {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});

describe("TokenComponent --- RTL", () => {
  it("Displays created tokens", async () => {
    global.rtlContextRouterRender(<TokenComponent {...props} />);
    expect(screen.queryAllByText(/Test User/i).length).toBeGreaterThan(0);
  });

  it("Open Create Modal", async () => {
    global.rtlContextRouterRender(<TokenComponent {...props} />);
    const button = await screen.findByTestId(/create-token-button/i);
    expect(screen.queryByText(/Create Global Token/i)).not.toBeInTheDocument();
    // await waitFor(() => fireEvent.click(button));
    fireEvent.click(button);
    expect(screen.getByText(/Create Global Token/i)).toBeInTheDocument();
  });
});
