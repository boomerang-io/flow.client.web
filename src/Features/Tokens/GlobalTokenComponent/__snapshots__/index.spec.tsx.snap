// Vitest Snapshot v1

exports[`TokenComponent --- Snapshot > Capturing Snapshot of TokenComponent 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="_container_23aca4"
    >
      <header
        class="cds--bmrg-feature-header _header_23aca4"
      >
        <section
          class="cds--bmrg-feature-header__content"
        >
          <div
            class="cds--bmrg-feature-header__header"
          >
            <h1
              class="cds--bmrg-feature-header-title _headerTitle_23aca4"
            >
              Global Tokens
            </h1>
            <p
              class="cds--bmrg-feature-header-title _headerTitle_23aca4"
            >
              Create tokens that can be used globally
            </p>
          </div>
        </section>
      </header>
      <div
        class="_content_23aca4"
      >
        <div
          class="_tokenContainer_23aca4"
        >
          <button
            class="cds--btn cds--btn--md cds--btn--primary"
            data-testid="create-token-button"
            style="width: 12rem;"
            tabindex="0"
            type="button"
          >
            Create Token
            <svg
              aria-hidden="true"
              aria-label="Create Token"
              class="cds--btn__icon"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
              />
            </svg>
          </button>
        </div>
        <div
          class="cds--data-table-container"
        >
          <div
            class="cds--data-table-content"
          >
            <table
              class="cds--data-table cds--data-table--sort"
            >
              <thead>
                <tr
                  class="_tableHeadRow_23aca4"
                >
                  <th
                    aria-sort="none"
                    class="_tableHeadHeader_23aca4 _creatorName_23aca4"
                    id="creatorName"
                    scope="col"
                  >
                    <div
                      id="table-sort-2"
                      style="display: none;"
                    >
                      Click to sort rows by Created By header in ascending order
                    </div>
                    <button
                      aria-describedby="table-sort-2"
                      class="_tableHeadHeader_23aca4 _creatorName_23aca4 cds--table-sort"
                      type="button"
                    >
                      <span
                        class="cds--table-sort__flex"
                      >
                        <div
                          class="cds--table-header-label"
                        >
                          Created By
                        </div>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon-unsorted"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                          />
                        </svg>
                      </span>
                    </button>
                  </th>
                  <th
                    aria-sort="none"
                    class="_tableHeadHeader_23aca4 _description_23aca4"
                    id="description"
                    scope="col"
                  >
                    <div
                      id="table-sort-3"
                      style="display: none;"
                    >
                      Click to sort rows by Description header in ascending order
                    </div>
                    <button
                      aria-describedby="table-sort-3"
                      class="_tableHeadHeader_23aca4 _description_23aca4 cds--table-sort"
                      type="button"
                    >
                      <span
                        class="cds--table-sort__flex"
                      >
                        <div
                          class="cds--table-header-label"
                        >
                          Description
                        </div>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon-unsorted"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                          />
                        </svg>
                      </span>
                    </button>
                  </th>
                  <th
                    aria-sort="descending"
                    class="_tableHeadHeader_23aca4 _creationDate_23aca4"
                    id="creationDate"
                    scope="col"
                  >
                    <div
                      id="table-sort-4"
                      style="display: none;"
                    >
                      Click to unsort rows by Date Created header
                    </div>
                    <button
                      aria-describedby="table-sort-4"
                      class="_tableHeadHeader_23aca4 _creationDate_23aca4 cds--table-sort cds--table-sort--active cds--table-sort--descending"
                      type="button"
                    >
                      <span
                        class="cds--table-sort__flex"
                      >
                        <div
                          class="cds--table-header-label"
                        >
                          Date Created
                        </div>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon-unsorted"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                          />
                        </svg>
                      </span>
                    </button>
                  </th>
                  <th
                    aria-sort="none"
                    class="_tableHeadHeader_23aca4 _expiryDate_23aca4"
                    id="expiryDate"
                    scope="col"
                  >
                    <div
                      id="table-sort-5"
                      style="display: none;"
                    >
                      Click to sort rows by Expires (UTC) header in ascending order
                    </div>
                    <button
                      aria-describedby="table-sort-5"
                      class="_tableHeadHeader_23aca4 _expiryDate_23aca4 cds--table-sort"
                      type="button"
                    >
                      <span
                        class="cds--table-sort__flex"
                      >
                        <div
                          class="cds--table-header-label"
                        >
                          Expires (UTC)
                        </div>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon-unsorted"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                          />
                        </svg>
                      </span>
                    </button>
                  </th>
                  <th
                    class="_tableHeadHeader_23aca4 _delete_23aca4"
                    id="delete"
                    scope="col"
                  />
                </tr>
              </thead>
              <tbody
                aria-live="polite"
                class="_tableBody_23aca4"
              >
                <tr>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="_tableCell_23aca4"
                    >
                      <p
                        class="_tableTextarea_23aca4"
                      >
                        Test User
                      </p>
                    </div>
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="_tableCell_23aca4"
                    >
                      <p
                        class="_tableTextarea_23aca4"
                      >
                        test token 2
                      </p>
                    </div>
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="_tableCell_23aca4"
                    >
                      <p
                        class="_tableTextarea_23aca4"
                      >
                        July 07, 2021
                      </p>
                    </div>
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="_tableCell_23aca4"
                    >
                      <p
                        class="_tableTextarea_23aca4"
                      >
                        ---
                      </p>
                    </div>
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="_tableCell_23aca4"
                    >
                      <button
                        aria-describedby="danger-description-7"
                        class="_button_188ad9 cds--btn cds--btn--sm cds--btn--danger--ghost"
                        data-testid="delete-token-button-12"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="cds--visually-hidden"
                          id="danger-description-7"
                        >
                          danger
                        </span>
                        <svg
                          aria-hidden="true"
                          aria-label="deleteToken"
                          class="cds--btn__icon"
                          fill="currentColor"
                          focusable="false"
                          height="16"
                          preserveAspectRatio="xMidYMid meet"
                          role="img"
                          viewBox="0 0 32 32"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M12 12H14V24H12zM18 12H20V24H18z"
                          />
                          <path
                            d="M4 6V8H6V28a2 2 0 002 2H24a2 2 0 002-2V8h2V6zM8 28V8H24V28zM12 2H20V4H12z"
                          />
                        </svg>
                      </button>
                    </div>
                  </td>
                </tr>
                <tr>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="_tableCell_23aca4"
                    >
                      <p
                        class="_tableTextarea_23aca4"
                      >
                        Test User
                      </p>
                    </div>
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="_tableCell_23aca4"
                    >
                      <p
                        class="_tableTextarea_23aca4"
                      >
                        test token
                      </p>
                    </div>
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="_tableCell_23aca4"
                    >
                      <p
                        class="_tableTextarea_23aca4"
                      >
                        June 07, 2021
                      </p>
                    </div>
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="_tableCell_23aca4"
                    >
                      <p
                        class="_tableTextarea_23aca4"
                      >
                        June 07, 2022
                      </p>
                    </div>
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="_tableCell_23aca4"
                    >
                      <button
                        aria-describedby="danger-description-8"
                        class="_button_188ad9 cds--btn cds--btn--sm cds--btn--danger--ghost"
                        data-testid="delete-token-button-11"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="cds--visually-hidden"
                          id="danger-description-8"
                        >
                          danger
                        </span>
                        <svg
                          aria-hidden="true"
                          aria-label="deleteToken"
                          class="cds--btn__icon"
                          fill="currentColor"
                          focusable="false"
                          height="16"
                          preserveAspectRatio="xMidYMid meet"
                          role="img"
                          viewBox="0 0 32 32"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M12 12H14V24H12zM18 12H20V24H18z"
                          />
                          <path
                            d="M4 6V8H6V28a2 2 0 002 2H24a2 2 0 002-2V8h2V6zM8 28V8H24V28zM12 2H20V4H12z"
                          />
                        </svg>
                      </button>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="cds--pagination cds--pagination--md"
        >
          <div
            class="cds--pagination__left"
          >
            <label
              class="cds--pagination__text"
              for="cds-pagination-select-id-14"
              id="cds-pagination-select-id-14-count-label"
            >
              Items per page:
            </label>
            <div
              class="cds--form-item cds--select__item-count"
            >
              <div
                class="cds--select cds--select--inline"
              >
                <div
                  class="cds--select-input--inline__wrapper"
                >
                  <div
                    class="cds--select-input__wrapper"
                  >
                    <select
                      class="cds--select-input"
                      id="cds-pagination-select-id-14"
                    >
                      <option
                        class="cds--select-option"
                        value="10"
                      >
                        10
                      </option>
                      <option
                        class="cds--select-option"
                        value="20"
                      >
                        20
                      </option>
                      <option
                        class="cds--select-option"
                        value="50"
                      >
                        50
                      </option>
                      <option
                        class="cds--select-option"
                        value="100"
                      >
                        100
                      </option>
                    </select>
                    <svg
                      aria-hidden="true"
                      class="cds--select__arrow"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="cds--pagination__text cds--pagination__items-count"
            >
              1–2 of 2 items
            </span>
          </div>
          <div
            class="cds--pagination__right"
          >
            <div
              class="cds--form-item cds--select__page-number"
            >
              <div
                class="cds--select cds--select--inline"
              >
                <label
                  class="cds--label cds--visually-hidden"
                  for="cds-pagination-select-id-14-right"
                >
                  Page number, of 1 pages
                </label>
                <div
                  class="cds--select-input--inline__wrapper"
                >
                  <div
                    class="cds--select-input__wrapper"
                  >
                    <select
                      class="cds--select-input"
                      id="cds-pagination-select-id-14-right"
                    >
                      <option
                        class="cds--select-option"
                        value="1"
                      >
                        1
                      </option>
                    </select>
                    <svg
                      aria-hidden="true"
                      class="cds--select__arrow"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="cds--pagination__text"
            >
              of 1 page
            </span>
            <div
              class="cds--pagination__control-buttons"
            >
              <span
                class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top cds--tooltip cds--icon-tooltip"
              >
                <button
                  aria-labelledby="tooltip-11"
                  class="cds--btn--icon-only cds--pagination__button cds--pagination__button--backward cds--pagination__button--no-index cds--btn cds--btn--md cds--btn--ghost cds--btn--disabled"
                  disabled=""
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M20 24L10 16 20 8z"
                    />
                  </svg>
                </button>
                <span
                  aria-hidden="true"
                  class="cds--popover"
                  id="tooltip-11"
                  role="tooltip"
                >
                  <span
                    class="cds--popover-content cds--tooltip-content"
                  >
                    Previous page
                  </span>
                  <span
                    class="cds--popover-caret"
                  />
                </span>
              </span>
              <span
                class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top-right cds--tooltip cds--icon-tooltip"
              >
                <button
                  aria-labelledby="tooltip-13"
                  class="cds--btn--icon-only cds--pagination__button cds--pagination__button--forward cds--pagination__button--no-index cds--btn cds--btn--md cds--btn--ghost cds--btn--disabled"
                  disabled=""
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 8L22 16 12 24z"
                    />
                  </svg>
                </button>
                <span
                  aria-hidden="true"
                  class="cds--popover"
                  id="tooltip-13"
                  role="tooltip"
                >
                  <span
                    class="cds--popover-content cds--tooltip-content"
                  >
                    Next page
                  </span>
                  <span
                    class="cds--popover-caret"
                  />
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
</body>
`;
